packages/core-contracts/src/contracts/arks/FluidLiteArk.sol

```diff
@@ -0,0 +1,289 @@
+                                INTERNAL FUNCTIONS
+    //////////////////////////////////////////////////////////////*/
+
+    /**
+     * @notice Internal function to get the total assets that are withdrawable
+     * @dev FluidLiteArk stores value in shares of the vault plus any WETH held directly
+     */
+    function _withdrawableTotalAssets()
+        internal
+        view
+        override
+        returns (uint256 withdrawableAssets)
+    {
+        // Get any currently held WETH
+        withdrawableAssets = config.asset.balanceOf(address(this));
+    }
+
+    /**
+     * @notice Deposits WETH into the vault through the router
+     * @dev Unwraps WETH to ETH first, then deposits via the router
+     * @param amount The amount of WETH to deposit
+     */
+    function _board(uint256 amount, bytes calldata) internal override {
+        // Unwrap WETH to ETH
+        weth.withdraw(amount);
+
+        // Submit ETH to StETH
+        steth.submit{value: amount}(address(this));
+        IERC20(address(steth)).forceApprove(address(vault), amount);
+        // this would fail if amount of steth is not enough
+        vault.deposit(amount, address(this));
+    }
+
+    /**
+     * @notice Withdraws assets from the vault
+     * @param amount The amount of WETH to withdraw
+     * @param  data Additional data (unused in this implementation)
+     */
+    function _disembark(uint256 amount, bytes calldata data) internal override {
+        // handled by the Ark.sol
+    }
+
+    /**
+     * @notice Requests a withdrawal from the withdrawal queue
+     * @param amount The amount of shares to withdraw
+     */
+    function requestWithdrawal(uint256 amount) external onlyKeeper {
+        if (withdrawalRequestId != 0) revert WithdrawalAlreadyRequested();
+        uint256 stethBalanceBefore = steth.balanceOf(address(this));
+
+        if (amount == type(uint256).max) {
+            uint256 shares = vault.maxRedeem(address(this));
+            vault.redeem(shares, address(this), address(this));
+        } else {
+            vault.withdraw(amount, address(this), address(this));
+        }
+        uint256 stethBalanceAfter = steth.balanceOf(address(this));
+
+        uint256[] memory amounts = new uint256[](1);
+        uint256 stethAmount = stethBalanceAfter - stethBalanceBefore;
+        amounts[0] = stethAmount;
+
+        IERC20(address(steth)).approve(address(withdrawalQueue), stethAmount);
+        uint256[] memory requestIds = withdrawalQueue.requestWithdrawals(
+            amounts,
+            address(this)
+        );
+
+        withdrawalRequestId = requestIds[0];
+
+        emit WithdrawalRequested(stethAmount, withdrawalRequestId);
+    }
+
+    /**
+     * @notice Requests a withdrawal from the withdrawal queue
+     * @param amount The amount of shares to withdraw
+     * @param data Additional data (unused in this implementation)
+     * @dev https://github.com/lidofinance/core/issues/442
+     */
+    function withdrawUsingSwap(
```

Comment:
Is it worth adding this to the IArkWithWithdrawalRequest interface given it's implemented on most of the arks. if unused, we just leave unimplemented, same as harvest for many arks

Reply:
it's there